Gemfile

```diff
@@ -28,12 +28,15 @@
 
 source 'https://rubygems.org'
 
-gem "rails", "~> 3.2.21"
+gem "rails", "4.0.13"
+gem 'protected_attributes'
+gem 'actionpack-action_caching'
+gem 'rails-observers'
```

Comment:
In this case I'm not sure that's really necessary. These gems restore deprecated Rails core functionality and do not add new features. As such, they don't have a meaningful, independent version number. I'd suggest being strict about the Rails version, and let Bundler dependency resolution take care of the rest.